conda/common/io.py

```diff
@@ -156,3 +164,70 @@ def attach_stderr_handler(level=WARN, logger_name=None, propagate=False, formatt
         logr.addHandler(new_stderr_handler)
         logr.setLevel(level)
         logr.propagate = propagate
+
+
+class ContextDecorator(object):
+    def __call__(self, f):
+        @wraps(f)
+        def decorated(*args, **kwds):
+            with self:
+                return f(*args, **kwds)
+        return decorated
+
+
+class time_recorder(ContextDecorator):  # pragma: no cover
+    start_time = None
+    record_file = expand(join('~', '.conda', 'instrumentation-record.csv'))
+
+    def __init__(self, entry_name):
+        self.entry_name = entry_name
+
+    def __enter__(self):
+        enabled = os.environ.get('CONDA_INSTRUMENTATION_ENABLED')
```

Comment:
Using a hard-coded environment variable here rather than putting it in `context` because `common` can't import from `conda.base`, and thus we don't have access to `context`.